dingo/bindings/hmc_sampling.h

```diff
@@ -0,0 +1,132 @@
+#include "bindings.h" 
+#include "ode_solvers/ode_solvers.hpp"
+#include <iostream> 
+#include <math.h> 
+#include <stdexcept> 
+using namespace std;
+
+
+// Initialization of the HPolytopeCPP class
+HPolytopeCPP::HPolytopeCPP() {}
+HPolytopeCPP::HPolytopeCPP(double *A_np, double *b_np, int n_hyperplanes, int n_variables){
+
+MT A;
+VT b;
+A.resize(n_hyperplanes,n_variables);
+b.resize(n_hyperplanes);
+
+int index = 0;
+for (int i = 0; i < n_hyperplanes; i++){
+   b(i) = b_np[i];
+   for (int j=0; j < n_variables; j++){
+      A(i,j) = A_np[index];
+      index++;
+   }
+}
+HP = Hpolytope(n_variables, A, b);                                 
+RNGType rng(HP.dimension());
+HP.normalize();
+int d = HP.dimension();
+   
+
+void hmc_leapfrog_gaussian(int walk_len,
+                                    int number_of_points, 
+                                    int number_of_points_to_burn, 
+                                    double* samples,
+                                    double variance,
+                                    double* bias_vector_,
+                                    double* inner_point) {
```

Comment:
Check the code inside if(method==10 { [this code] } else if()....

We want to run only this code in function `hmc_leapfrog_gaussian()` and nothing else. This means that the input of the function must be exactly what we need to know inside the `if(method==10) {..}` and the output must be the `list<Point> rand_points`, which is what the function `apply_sampling()` needs to continue running after calling `hmc_leapfrog_gaussian()`